packages/evm/contracts/BoostCore.sol

```diff
@@ -204,6 +227,46 @@ contract BoostCore is Ownable, ReentrancyGuard {
         return _boosts.length;
     }
 
+    /// @notice Get the incentives for a Boost
+    /// @param key The key composed of the Boost ID and the Incentive ID - keccak256(abi.encodePacked(boostId, incentiveId))
+    function getIncentive(bytes32 key) external view returns (IncentiveDisbursalInfo memory) {
+        return incentives[key];
+    }
+
+    /// @notice Returns the protocol fee and any remaining incentive value to the owner or budget
+    /// @param boostId The ID of the Boost
+    function clawback(bytes calldata data_, uint256 boostId, uint256 incentiveId) external nonReentrant {
```

Comment:
How is this going to be permissioned? Is it going to be an admin function? I don't see a viable story where a boost creator would call into this. In this current state anyone can drain an incentive since ClawbackPayload includes an address to send the funds to.

Reply:
It's going to be permissioned based on the incentive permissions you worked on (i.e. isAuthorized on Incentive) comment bellow elludes to that.